### ISSUE TITLE
Antialiasing functionality doesn't do anything

### ISSUE THREAD

**Issue by mvl22** [nptscot/nptscot.github.io#136]
As far as I can see, the antialiasing control doesn't actually do anything, because the toggle simply reloads the page, but no cookie is set so the state upon reload is always back to the default off.

@mem48 can you clarify whether this does actually work? If so, can you give me a clear reproduce case?

**Comment by mem48** [comment]
On https://npt.scot/#14/55.94026/-3.19659

Off:

![image](https://github.com/nptscot/nptscot.github.io/assets/15819577/666022a6-f994-4123-b52e-7d43c23a1668)

On:

![image](https://github.com/nptscot/nptscot.github.io/assets/15819577/df1f5576-0350-43ca-86a7-4739babbb289)

**Comment by mem48** [comment]
So seem to work to me. Notice the jagged edges on the buildings above Thirlestane Road

**Comment by mem48** [comment]
There is a

```
var antialias = document.getElementById('antialiascheckbox').checked;
console.log('Antialias is ' + antialias);
```

**Comment by mem48** [comment]
Toggling the switch triggers a reload, which reloads the map but does not change the state of the switch. So once the map loads it loads it with the new setting.

**Comment by mvl22** [comment]
Thanks. But I'm still struggling either to reproduce this or understand how it could work, I'm afraid.

The reload is done using:

```
function switch_antialias(){
  location.reload()
};
```

`location.reload()` reloads the current URL in the URL bar, i.e. the entire page. That should mean that the entire state is reset entirely, meaning any change to the form control will be lost on the reloaded page.

So the process going on seems to be:

- Site is loaded
- User clicks the antialias switch
- The page is reloaded
- Page is back to initial state, which has anti-aliasing off as there is no cookie to remember the form control state before the reload.

I can't see any point at which the state actually has the anti-aliasing kicking in.

#70 does indeed mention the need for a cookie-based control, as I would expect, but the commit referenced has no cookie implementation.

Can you confirm your browser is not using any plugins which try to maintain form state to avoid losing data? Does the anti-aliasing that you are seeing working work using a different browser?

**Comment by mem48** [comment]
Ah, I think I found the problem. On firefox an F5 refresh does not change the state of forms. At least on my computer. So all the UI settings are retained (e.g.change the trip type to "Primary School", refresh and the setting remains). I've just tested this on Chrome and it does not work (All setting revert to defaults). So by the same logic the antialiasing feature only works on Firefox.

**Comment by mem48** [comment]
So you are right, we need some cookie to store the state of the toogle across refreshes. I'm surpised I hadn't noticed that difference between browsers, or that none of the other UI elements are broken because of it. I just assumed that how all browsers worked.

**Comment by mem48** [comment]
There is some pre-exiting cookie logic for the Google Analytics opt-out. So you may want to reuse inprove on that.

**Comment by mvl22** [comment]
> On firefox an F5 refresh does not change the state of forms. At least on my computer.

Ah, OK this is making sense now. Seems like this is by design on Firefox but not other browsers:

https://bugzilla.mozilla.org/show_bug.cgi?id=1255592

I'll implement cookie state shortly.

**Comment by mvl22** [comment]
> I'll implement cookie state shortly.

Fixed in 6c679581731e00f7719756c02b80f61e366cd250.

**Comment by mvl22** [comment]
Anti-aliasing is now set to be on by default on desktop, in 0ddd6e8e6803c75a5edf96af23a326aa7d0a9ec6.
